src/core/ucc_constructor.c

```diff
@@ -26,7 +27,8 @@ static int callback(struct dl_phdr_info *info, size_t size, void *data)
         component_path = (char *)ucc_malloc(pos + UCC_COMPONENT_LIBDIR_LEN + 1,
                                             "component_path");
         if (!component_path) {
-            //TODO add ucc_error(msg) when logging as added
+            ucc_error("failed to allocate %d bytes for component_path",
+                      pos + UCC_COMPONENT_LIBDIR_LEN + 1);
```

Comment:
return UCC_ERR_NO_MEMORY; ?

Reply:
this "callback" function is part of libdl interface - it has to return "int", not ucc_status_t.